kolibri/auth/models.py

```diff
@@ -242,14 +233,44 @@ def remove_user(self, user, role_kind):
         """
 
         # ensure the specified role kind is valid
-        assert role_kind in (kind[0] for kind in Role.KINDS), \
+        assert role_kind in (kind[0] for kind in role_kinds.choices), \
             "'{role_kind}' is not a valid role kind.".format(role_kind=role_kind)
 
         # ensure the provided user is a FacilityUser
         assert isinstance(user, FacilityUser), "Only FacilityUsers can be associted with a collection."
 
         # delete the appropriate role, if it exists
-        Role.objects.filter(user=user, collection=self, kind=role_kind, dataset=self.dataset).delete()
+        Role.objects.filter(user=user, collection=self, kind=role_kind).delete()
+
+    def add_member(self, user):
+        """
+        Create a Membership associating the provided user with this collection.
+        If the Membership object already exists, just return that, without changing anything.
+
+        :param user: The FacilityUser to add to this Collection.
+        :return: The Membership object (possibly new) that associates the user with the Collection.
+        """
+
+        # ensure the provided user is a FacilityUser
+        assert isinstance(user, FacilityUser), "Only FacilityUsers can be members of a collection."
+
+        # create the necessary membership, if it doesn't already exist
+        membership, created = Membership.objects.get_or_create(user=user, collection=self)
+
+        return membership
+
+    def remove_member(self, user):
+        """
+        Remove any Membership objects associating the provided user with this collection.
+
+        :param user: The FacilityUser to remove from this Collection.
+        """
+
+        # ensure the provided user is a FacilityUser
+        assert isinstance(user, FacilityUser), "Only FacilityUsers can be removed from a collection."
+
+        # delete the appropriate membership, if it exists
+        Membership.objects.filter(user=user, collection=self).delete()
```

Comment:
Actually, since there shouldn't ever be more than one deleted, and no related models should be deleted, it'll be cleaner to just return True/False to indicate whether anything was deleted.